### PR TITLE
[TD-1044]: Update sync-automation to pick commits.

### PR DESCRIPTION
**Pull request by asutosh**
## Description
These changes ~~enable picking entire commits in case of
changes to code contained in the `ci/` directory.~~
remove the current files/directories tracked by sync-automation and restores them from the master branch(when it's triggered by changes to any of these tracked files/directories on pushes to master branch)
More details in TD-1044

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4218/merge
Commit:  
Triggered by: pull_request (@asutosh)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2802385137)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4218/merge
Commit: 4e4649679687a31687c8eb8b0f8b860f6b1235ce 
Triggered by: pull_request (@asutosh)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2815727791)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4218/merge
Commit: f92bd4ea3d8f1174ede47d422e169a661c50c0aa 
Triggered by: pull_request (@asutosh)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2834635409)

File: .github/workflows/sync-automation.yml [hunk 1]
```diff
@@ -41,13 +38,46 @@ jobs:
...
           git config --local user.name "Bender"
-          for f in $FILES; do 
-            git checkout --theirs origin/master -- $f || true
-          done
-          git commit -a -m "[CI] Syncing release automation from master"
-          git push origin
-          echo "::debug::${FILES} syncd for ${{matrix.branch}}"
+          git fetch origin ${{ matrix.branch }}
+          prbranch=sync/auto/${{ matrix.branch }}/${{ github.run_id }}
+          git checkout -b $prbranch
```
Comment (alephnull) [inline .github/workflows/sync-automation.yml:47]: Better to parametrise this. Unless it is a template variable.

**Review by alephnull — APPROVED** [review]

File: .github/workflows/sync-automation.yml (hunk 1 above)
Comment (asutosh) [reply .github/workflows/sync-automation.yml:47]: Was thinking of making them template vars.  The same set of files are used in three different places - the trigger condition, deletion, and restore. These might also differ from repo to repo. We could rewrite the template accordingly, so it's handled there.

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4218/merge
Commit: d6ef1f6fa25a17bf5faf926fcaa133baeb015293 
Triggered by: pull_request (@asutosh)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2838509306)

**Review by ermirizio — APPROVED** [review]
Looks fine for me. Lets get this merge and see how it goes.

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4218/merge
Commit: ad838605c3f65deb481bf6c6c0aba8293e8d35a1 
Triggered by: pull_request (@asutosh)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2844920239)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/heads/master
Commit: 91db2aa60650b92c6068980add1d1825eddf5bde [TD-1044]: Update sync-automation. (#4218)

* Update sync-automation.

Based on the discussions on TD-1044, for sync-automation, we 
 delete tracked files/directories and restore from master branch - and 
then create PR from the changes to the target release branches.

[changelog]
internal: Update sync-automation to create PRs.
Triggered by: push (@asutosh)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2845074745)

**Pull request merged**